### PR TITLE
Working draft of the plotting tutorial

### PR DESCRIPTION
**Pull request by mistShard**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [x] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue #201 
- Closes #

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

## If you have comments or can explain your changes, please do so below.
I added two files to aid in creating plotting tutorials. The tutorial shows a single graph now.

### CODE REVIEW

File: docs/basics/plotting.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        plt.legend()
+
+        max_population = pd_df['Global Population'].max()
+        max_year = pd_df.loc[pd_df['Global Population'].idxmax(), 'Year']
+        unit = 1000000000
+
+        plt.annotate(f'Max: {max_population/unit:.2f}B', xy=(max_year, max_population), 
+                    xytext=(-90, -10), textcoords='offset points', ha='left', va='bottom',
+                    bbox=dict(boxstyle='round,pad=0.5', fc='blue', alpha=0.2),
+                    arrowprops=dict(arrowstyle='->', connectionstyle='arc3,rad=0'))
```
Comment (MarcoGorelli) [inline docs/basics/plotting.md:121]: it looks like here this is specific to pandas

can we make a dataframe-agnostic function which creates a plot?

something like
```
def plot_population(df_any):
    df = nw.from_native(df_any, eager_only=True)
    # plotting logic, produce plot
```

**Review by MarcoGorelli — COMMENTED** [review]
thanks for working on this!

File: docs/basics/plotting.md (hunk 1 above)
Comment (mistShard) [reply docs/basics/plotting.md:121]: Okay. To be sure I understand, you are asking that I use the result of `nw.from_native(df_any, eager_only=True` to create a plot within the `plot_population` function?

Comment (MarcoGorelli) [reply docs/basics/plotting.md:121]: yup

File: docs/basics/plotting.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    └────────┴───────────────────┘
+    ```
+
+We can then use matplotlib to plot a graph of the Global population trend from the year 1555 till 2021. To do so, we need to create a dataframe-agnostic Narwhals function.
+
+```python
+def plot_population_line(df_any):
+    df = nw.from_native(df_any)
+    plt.figure(figsize=(10, 5), layout = 'constrained', )
+    plt.plot(df['Year'], df['Global Population'], label = 'Population Growth', linewidth=1, marker='o', markersize=2, color='purple')
```
Comment (MarcoGorelli) [inline docs/basics/plotting.md:111]: shall we explicitly convert to numpy here? that's what matplotlib would do under the hood anyway
```suggestion
    plt.plot(df['Year'].to_numpy(), df['Global Population'].to_numpy(), label = 'Population Growth', linewidth=1, marker='o', markersize=2, color='purple')
```

**Pull request closed without merging**